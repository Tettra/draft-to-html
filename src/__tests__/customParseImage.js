import { convertFromRaw } from 'draft-js'
import convertToHtml from '../'
import image from './image.json'

const parseBlock = (block, contentState) => {
  if (block.getType().toLowerCase() === 'atomic') {
    const entity = contentState.getEntity(block.getEntityAt(0))
    if (entity != null && entity.getType() === 'IMAGE') {
      const { src } = entity.getData()
      const el = document.createElement('img')

      el.setAttribute('href', src)
      el.setAttribute('alt', name)
      el.textContent = name
      return el
    }
  }
}

describe('convertToHtml with custom parseBlock option', () => {
  test('converts image with custom parseBlock function', () => {
    const html = convertToHtml(convertFromRaw(image), { parseBlock })
    expect(html).toMatchSnapshot()
  })
})
