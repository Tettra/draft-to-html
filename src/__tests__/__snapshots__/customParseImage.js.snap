// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToHtml with custom parseBlock option converts image with custom parseBlock function 1`] = `"<img>"`;
